adapters/handlers/rest/handlers_authz.go

```diff
@@ -30,11 +30,14 @@ type authZHandlers struct {
 }
 
 type authzController interface {
```

Comment:
can we not add abstractions when we don't need them? The interface is just a copy of all public method of the only implementation

Reply:
Hmm, is it not a good idea at this stage to define the dependency as an interface so that it is easier to mock it out at a later date when testing `authZHandlers` methods?